apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java

```diff
@@ -2379,6 +2382,100 @@ static void mapEobCommonGroupInpHHAHospiceSNFCoinsurance(
     }
   }
 
+  /**
+   * Generically attempts to retrieve a diagnosis from the current claim.
+   *
+   * @param substitution The methods to retrive diagnosis information all follow a similar pattern.
+   *     This value is used to substitute into that pattern when looking up the specific method to
+   *     retrive information with.
+   * @param claim Passed as an Object because there is no top level `Claim` class that claims derive
+   *     from
+   * @param ccw CCW Codebook value that represents which "PresentOnAdmissionCode" is being used.
+   *     Example: {@link CcwCodebookVariable#CLM_POA_IND_SW5}
+   * @param label One or more labels to use when mapping the diagnosis.
+   * @return a {@link Diagnosis} or {@link Optional#empty()}
+   */
+  public static Optional<Diagnosis> extractDiagnosis(
+      String substitution, Object claim, Optional<CcwCodebookInterface> ccw, DiagnosisLabel label) {
+
+    Optional<String> code =
+        ReflectionUtils.tryMethod(claim, String.format("getDiagnosis%sCode", substitution));
```

Comment:
This code is from DiagnosisUtilsV2, which is how this is implemented in V2. Im having trouble modifying the existing code to get this to work in V1 without reflection. I say maybe since the reflection code is only being used to extract procedures/diagnoses, we can have a separate ticket related to removing reflection completely from the codebase.

Reply:
Since it's an existing piece of code, created https://jira.cms.gov/browse/BFD-2343 to handle this and other instances of it and will not block this PR.